### ISSUE TITLE
Fix reversal of gate-application modifiers

### ISSUE THREAD

**Issue by appleby** [rigetti/quilc#317]
Something in the `PARSE-QUIL` -> `PRINT-PARSED-PROGRAM` cycle causes sequences of `CONTROLLED DAGGER` or `DAGGER CONTROLLED` to permute. For example:

```
CONTROLLED DAGGER Y 0 1
-> parse -> print ->
DAGGER CONTROLLED Y 0 1
-> parse -> print ->
CONTROLLED DAGGER Y 0 1
```

This does not affect correctness, but is mildly annoying and does affect the "fixed-point" check of `TEST-PRINT-PARSED-PROGRAM-GOLDEN-FILES`.

We should either print these modifiers in the same order they appear in the input Quil program, or else pick some canonical ordering so they are always printed the same way.

Once implemented, be sure to re-enable the fixed-point check for the test in `tests/printer-test-files/gold-standard/plundered-gold.quil` which is reference in the following code review discussion thread:

https://github.com/rigetti/quilc/pull/316#discussion_r300177587

**Comment by appleby** [comment]
This issue is (temporarily) undead. #319

**Comment by appleby** [comment]
I updated the issue title since this applies to FORKED modifiers as well. For example

	./quilc <<EOF
    > DEFCIRCUIT MODIFIERS q0 q1 q2:
    >	FORKED CONTROLLED DAGGER Y q0 q1 q2
    > HALT
    > EOF
    DEFCIRCUIT MODIFIERS q0 q1 q2:
	    DAGGER CONTROLLED FORKED Y q0 q1 q2

    HALT

**Comment by appleby** [comment]
This was fixed in #329
